crates/holochain/src/core/ribosome/host_fn/must_get_agent_activity.rs

```diff
@@ -57,6 +57,11 @@ pub fn must_get_agent_activity(
 
                 let result: Result<_, RuntimeError> = match (result, &call_context.host_context) {
                     (Activity(activity), _) => Ok(activity),
+                    (Warrants(_), _) => {
+                        // TODO: we probably want to handle this differently
```

Comment:
i.e., return the warrant instead of nothing, but that would be a breaking api change

Reply:
Maybe this is exactly right though. Because must_get_agent_activity needs to be deterministic, and warrants can be different, but all should result in no valid data being returned.

Reply:
> Yesterday you said that in one moment there can be no warrants and in the next there can be some. What is the difference to other agent activity?

must_get_agent_activity queries are always hash-bounded, so even if the activity grows, the growth will be beyond the bounds of any particular query. Not so for warrants, which can show up for any hash range at any time.

Reply:
Right, but will it be deterministic in the sense that when two agents run the same hash bounded must_get_agent_activity query, they'll get the same warrants?

Reply:
Nope -- race conditions and network conditions like partitions could lead to different results at different times, which is not the case for the actions themselves